### DISCUSSION TITLE
v2 rest api for departures missing data?

### DISCUSSION THREAD

**Discussion by DaveNash** [transitland/transitland#433] [Q&A] [answered]
I'm getting almost no data for Caltrain or BART at the moment.
E.g.: 
https://transit.land/api/v2/rest/stops/s-9q9hxhe7jx-sunnyvalecaltrainstationsouthbound/departures?apikey=[xxx]&service_date=2024-08-07 
returns 2 departures
or 
https://transit.land/api/v2/rest/stops/s-9q9p1wxcfk-macarthur/departures?apikey=[xxx]&service_date=2024-08-07
returns 20 departures (all between 1pm and 2pm)

I can't see anything wrong on my end, though that's always a possibility. Please advise.

**Comment by dinwun450** [comment]
I have the same problem, too. Is Transitland's departures offline for the Bay Area, or is the 511 source for transit departures unavailable?!

**Comment by irees** [comment]
hi @DaveNash @dinwun450  - I am looking into this issue today.

**Comment by irees** [comment, marked as answer]
@DaveNash @dinwun450 - Can you see if the problem is resolved now? 

There was a code change that changed the behavior when "service_date" was provided but "start_time/end_time" was not provided. In this case, it would default to "next=3600", e.g. the next 1 hour worth of departures. It's now been fixed to use the previous behavior, where "service_date" defaults to "all departures on that date". You can still use "start_time/end_time" and "next" parameters to set explicit time requirements for the selected service day.

**Reply by DaveNash** [reply]
Hey, Ian. Seems to be working now, though it took me some playing with the 'limit' query parameter to confirm it. Looks like limit defaults to 20 (I think?), which isn't documented. Is there a max for 'limit'? I tried setting it to 1000, and got back 329 departures for one BART station, so I'm probably fine. 
To me, it seems like 'limit' should default to all for the other params, in keeping with your change to 'start' and 'end time' above. 
Thanks for your support, as well as the new param options!

**Reply by janellehlee** [reply]
I'm having trouble loading some stops for BART. I get `{'error': 'the requested element is null which the schema does not allow'}`
[When looking up stops on the site](https://www.transit.land/stops/s-9q8yywd0c2-powellstreet#departures), I also see "ApolloError: the requested element is null which the schema does not allow"

Here's a list of the stops affected:
```
 ['s-9q8yywd0c2-powellstreet', 's-9q9nfsxnkk-fruitvale', 's-9q9pb85d1u-elcerritodelnorte', 's-9q9p1wxcfk-macarthur',       
 's-9q9ng2g43f-coliseum', 's-9q9p19umpw-12thstreet~oaklandcitycenter', 's-9q9p1dwqm2-19thstreetoakland', 's-9q8yyzcnqh-embarcadero', 's-9q8yymscp0-civiccenter~unplaza', 's-9q8yyxq436-montgomerystreet']
```

**Reply by irees** [reply]
@jhlee-cs I'm looking into this.

**Reply by drewda** [reply]
FYI for @jhlee-cs, this issue has now been fixed.
